test_coverage.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# Prerequisites
+# 1. You need LLVM-COV tools:
+# $ rustup component add llvm-tools-preview
+# 2. and Rust wrappers for llvm-cov:
+# $ cargo install cargo-binutils
+# 3. The rust name demangler
+# $ cargo install rustfilt
+# 4. jq
+# 5. genhtml
+# $ sudo apt install lcov
+
+RUSTFLAGS="-Z instrument-coverage"
+LLVM_PROFILE_FILE="./cov_raw/tari_crypto-%m.profraw"
+
+get_binaries() {
+  files=$( RUSTFLAGS=$RUSTFLAGS cargo test --tests --no-run --message-format=json \
+              | jq -r "select(.profile.test == true) | .filenames[]" \
+              | grep -v dSYM - \
+        );
+  files=("${files[@]/#/-object }")
+}
+
+get_binaries
+echo $files
+
+RUSTFLAGS=$RUSTFLAGS LLVM_PROFILE_FILE=$LLVM_PROFILE_FILE cargo test --tests
+
+cargo profdata -- \
+  merge -sparse ./cov_raw/tari_crypto-*.profraw -o ./cov_raw/tari_crypto.profdata
+
+cargo cov -- \
+  export \
+    --Xdemangler=rustfilt \
+    --format=lcov \
+    --show-branch-summary \
+    --show-instantiation-summary \
+    --show-region-summary \
+    --ignore-filename-regex='/.cargo/registry' \
+    --instr-profile=cov_raw/tari_crypto.profdata \
+    $files \
+    > cov_raw/tari_crypto.lcov
+
+if [ -z ${SKIP_HTML+x} ]; then
+  genhtml -o coverage_report cov_raw/tari_crypto.lcov
```

Comment:
I got this error in the report generation on mac:

`genhtml: ERROR: cannot read /rustc/657bc01888e6297257655585f9c475a0801db6d2/library/std/src/thread/local.rs`

I added `--ignore-filename-regex="^/rustc" \` to the `cargo cov` command  which worked - not sure if that is fixing a symptom though.

Reply:
Thanks; I think that exclusion makes sense.